includes/Core/User_Input/User_Input.php

```diff
@@ -228,8 +227,24 @@ public function set_answers( $settings ) {
 			$setting_data['scope']  = static::$questions[ $setting_key ]['scope'];
 
 			if ( 'site' === $setting_data['scope'] ) {
-				$setting_data['answeredBy']    = $this->user_options->get_user_id();
+				$existing_answers = $this->get_answers();
+				$answered_by      = $this->user_options->get_user_id();
+				if ( ! empty( $existing_answers['purpose']['answeredBy'] ) ) {
+					// If answers are not saved for the first time, keep original attribution.
+					$answered_by = $existing_answers['purpose']['answeredBy'];
+				}
+
+				if ( ! empty( $existing_answers['purpose']['values'] ) ) {
+					$answer_differs = array_diff( $existing_answers['purpose']['values'], $answers );
+					// If purpose answer changed, attribute it to the current user.
+					if ( ! empty( $answer_differs ) ) {
+						$answered_by = $this->user_options->get_user_id();
+					}
+				}
+
+				$setting_data['answeredBy']    = $answered_by;
```

Comment:
I know the logic here makes sense, but I find it a bit tough to follow. In part because we assign `$answered_by` a default that should really only be the default when there's no answer given.

What do you think of this change? Or something like it? Just to follow the logic a bit more. Let me know 🙂 

```suggestion
				if (
					// If the answer to the "purpose" question changed,
					// attribute the answer to the current user changing the
					// answer.
					(
						! empty( $existing_answers['purpose']['values'] ) &&
						! empty( array_diff( $existing_answers['purpose']['values'], $answers ) )
					) ||
					// If the answer to the "purpose" question was empty,
					// attribute the answer to the current user.
					empty( $existing_answers['purpose']['answeredBy'] )
				) {
					$answered_by = $this->user_options->get_user_id();
				} else {
					// Otherwise, attribute the answer to the user who answered
					// the question previously.
					$answered_by = $existing_answers['purpose']['answeredBy'];
				}

				$setting_data['answeredBy'] = $answered_by;
```

Reply:
Thanks, this looks fine, with more explanations along the way, I will commit the change